quarkus-integration-test-kafka build fails in native mode with JDK 11

**Describe the bug**
```
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:native-image (native-image) @ quarkus-integration-test-kafka ---
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /Users/Gwenneg/git/quarkus/integration-tests/kafka/target/quarkus-integration-test-kafka-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-kafka-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /Users/Gwenneg/git/quarkus/integration-tests/kafka/target/quarkus-integration-test-kafka-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-kafka-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 19.3.0 CE
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] docker run -v /Users/Gwenneg/git/quarkus/integration-tests/kafka/target/quarkus-integration-test-kafka-999-SNAPSHOT-native-image-source-jar:/project:z --rm quay.io/quarkus/ubi-quarkus-native-image:19.3.0-java11 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -H:+TraceClassInitialization --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar quarkus-integration-test-kafka-999-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+AddAllCharsets -H:EnableURLProtocols=http -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-kafka-999-SNAPSHOT-runner
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]    classlist:  60,944.83 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]        (cap):   5,606.92 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]        setup:  13,967.52 ms
20:15:52,004 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Final
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]   (typeflow): 163,639.69 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]    (objects):  98,546.11 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]   (features):   5,271.64 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]     analysis: 282,027.29 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]     (clinit):   6,708.41 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]     universe:  29,754.20 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]      (parse):  37,709.44 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]     (inline):  44,072.53 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]    (compile): 183,881.50 ms
[quarkus-integration-test-kafka-999-SNAPSHOT-runner:24]      compile: 267,364.97 ms
Fatal error: org.graalvm.compiler.debug.GraalError: java.lang.NullPointerException
	at method: Checksum org.apache.kafka.common.utils.Crc32C$Java9ChecksumFactory.create()  [Direct call from long DefaultRecord.computePartialChecksum(long, int, int)]
	at com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:991)
	at com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:922)
	at com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:259)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:171)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: java.lang.NullPointerException
	at com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:937)
	... 9 more
Error: Image build request failed with exit status 1
```

**To Reproduce**
Steps to reproduce the behavior from the `master` branch:
1. (Optional) Set the `JAVA_HOME` environment variable to a JDK 11 installation directory (OpenJDK 11 or GraalVM 19.3.0 JDK 11 edition). Add the JDK 11 `bin` directory to the path.
2. (Optional) Run `./mvnw clean install -Dmaven.test.skip=true` to build all the Quarkus sources with the JDK 11.
3. Run `./mvnw clean install --projects :quarkus-integration-test-kafka -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:19.3.0-java11`.

--- a/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java
+++ b/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java
@@ -2,7 +2,6 @@ package io.quarkus.kafka.client.deployment;
 
 import java.util.Arrays;
 import java.util.Collection;
-import java.util.zip.Checksum;
 
 import org.apache.kafka.clients.consumer.RangeAssignor;
 import org.apache.kafka.clients.consumer.RoundRobinAssignor;
@@ -34,17 +33,11 @@ import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
 
 import io.quarkus.deployment.Capabilities;
-import io.quarkus.deployment.GeneratedClassGizmoAdaptor;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
-import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
 import io.quarkus.deployment.builditem.JniBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
-import io.quarkus.gizmo.ClassCreator;
-import io.quarkus.gizmo.ClassOutput;
-import io.quarkus.gizmo.MethodCreator;
-import io.quarkus.gizmo.MethodDescriptor;
 import io.quarkus.kafka.client.serialization.JsonbDeserializer;
 import io.quarkus.kafka.client.serialization.JsonbSerializer;
 import io.quarkus.kafka.client.serialization.ObjectMapperDeserializer;
@@ -75,7 +68,6 @@ public class KafkaProcessor {
             StringDeserializer.class,
             FloatDeserializer.class,
     };
-    static final String TARGET_JAVA_9_CHECKSUM_FACTORY = "io.quarkus.kafka.client.generated.Target_Java9ChecksumFactory";
 
     @BuildStep
     public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
@@ -114,30 +106,5 @@ public class KafkaProcessor {
 
         // enable JNI
         jni.produce(new JniBuildItem());
-    }
-
-    /**
-     * Generate a class which replaces the usage of {@code MethodHandle} in {@code Java9ChecksumFactory} with a plain
-     * constructor invocation when run under GraalVM. This is necessary because the native image generator does not
-     * support method handles.
-     *
-     * @return the generated class
-     */
-    @BuildStep
-    public void replaceJava9Code(BuildProducer<GeneratedClassBuildItem> producer) {
-        // make our own class output to ensure that our step is run.
-        ClassOutput classOutput = new GeneratedClassGizmoAdaptor(producer, false);
-        try (ClassCreator cc = ClassCreator.builder().className(TARGET_JAVA_9_CHECKSUM_FACTORY)
-                .classOutput(classOutput).setFinal(true).superClass(Object.class).build()) {
-
-            cc.addAnnotation("com/oracle/svm/core/annotate/TargetClass").addValue("className",
-                    "org.apache.kafka.common.utils.Crc32C$Java9ChecksumFactory");
-            cc.addAnnotation("com/oracle/svm/core/annotate/Substitute");
-
-            try (MethodCreator mc = cc.getMethodCreator("create", Checksum.class)) {
-                mc.addAnnotation("com/oracle/svm/core/annotate/Substitute");
-                mc.returnValue(mc.newInstance(MethodDescriptor.ofConstructor("java.util.zip.CRC32C")));
-            }
-        }
     }
 }

--- a/extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/graal/Crc32CSubstitutions.java
+++ b/extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/graal/Crc32CSubstitutions.java
@@ -1,0 +1,35 @@
+package io.quarkus.kafka.client.runtime.graal;
+
+import java.lang.invoke.MethodHandle;
+import java.util.zip.Checksum;
+
+import org.apache.kafka.common.utils.Crc32C;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.RecomputeFieldValue.Kind;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK11OrLater;
+
+/**
+ * The following substitution replaces the usage of {@code MethodHandle} in {@code Java9ChecksumFactory} with a plain
+ * constructor invocation when run under GraalVM. This is necessary because the native image generator does not support method
+ * handles.
+ */
+@TargetClass(value = Crc32C.class, innerClass = "Java9ChecksumFactory", onlyWith = JDK11OrLater.class)
+final class Target_org_apache_kafka_common_utils_Crc32C_Java9ChecksumFactory {
+
+    @Alias
+    @RecomputeFieldValue(kind = Kind.Reset)
+    private static MethodHandle CONSTRUCTOR;
+
+    @Substitute
+    public Checksum create() {
+        try {
+            return (Checksum) Class.forName("java.util.zip.CRC32C").getConstructor().newInstance();
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}

